Jenkinsfile

```
@@ -0,0 +1,281 @@
+    BAHTTP_PROXY = "${env.HTTP_PROXY ? '--build-arg HTTPS_PROXY="' + env.HTTPS_PROXY + '" --build-arg https_proxy="' + env.HTTPS_PROXY + '"' : ''}"
+    UID=sh(script: "id -u", returnStdout: true)
+    BUILDARGS = "--build-arg NOBUILD=1 --build-arg UID=$env.UID $env.BAHTTP_PROXY $env.BAHTTPS_PROXY"
+    DOCKERFILE = "${params.DOCKERFILE ? params.DOCKERFILE : 'Dockerfile.centos.7'}"
+    FUSE_COMMIT = '3e2fcf3a630e575bc420df254525834504dc01b7'
+    HWLOC_COMMIT = 'refs/tags/hwloc-1.11.5'
+    MERCURY_COMMIT = '674e7f2bd17b5d8b85606cd152dd1bc189899b0e'
+    OFI_COMMIT = '8c33f9d63d536cc3781017dd25b7bb480ac96cb5'
+    OMPI_COMMIT = '373098d8ae0053af85cc1d49a58dbe933a31a50a'
+    OPENPA_COMMIT = '8e1e74feb22d2e733f34a96e6c7834fed3073c52'
+    PMIX_COMMIT = '8c2ffe7e6837a2fb76b882e4c5765032b2b84fa9'
+  }
+
+  options {
+    // preserve stashes so that jobs can be started at the test stage
+    preserveStashes(buildCount: 5)
+    checkoutToSubdirectory('scons_local')
+  }
+
+  stages {
+    stage('Phase 1') {
+    parallel {
+      stage ('code_lint_check') {
+
+        // Just use CentOS for lint checks.
+        agent {
+           dockerfile {
+             filename "${env.DOCKERFILE}"
+             dir 'scons_local/docker'
+             label 'docker_runner'
+             additionalBuildArgs '$BUILDARGS'
+           }
+        } // agent
+        steps {
+          checkPatch review_creds: "${env.GITHUB_CREDS}"
+        } // steps
+      } // stage ('code_lint_check')
+      stage ('fuse build') {
+        agent {
+          dockerfile {
+            filename "${env.DOCKERFILE}"
+            dir 'scons_local/docker'
+            label 'docker_runner'
+            additionalBuildArgs '$BUILDARGS'
+          }
+        } // agent
+        steps {
+          // Older scons_local_review only used master branch.
+          // Newer one looked up the last known good build of master branch.
+          // Pipeline currently does not have access that info.
+
+          sh 'mkdir -p testbin; rm -rf testbin/*'
+          sconsBuild target: 'fuse',
+                     directory: 'scons_local',
+                     scm: [url: 'https://github.com/libfuse/libfuse.git',
+                           branch: "${env.FUSE_COMMIT}",
+                           cleanAfterCheckout: true],
+                     no_install: true,  // No separate install step
+                     TARGET_PREFIX: '/testbin',
+                     target_work: 'testbin'
+
+          echo "fuse build succeeded"
+        } // steps
+      } // stage ('fuse build')
+      stage ('openpa prebuild') {
+        agent {
+          dockerfile {
+            filename "${env.DOCKERFILE}"
+            dir 'scons_local/docker'
+            label 'docker_runner'
+            additionalBuildArgs '$BUILDARGS'
+          }
+        } // agent
+        steps {
+          sh 'mkdir -p testbin; rm -rf testbin/*'
```

This would actually be better the other way around:

```suggestion
          sh 'rm -rf testbin/ && mkdir -p testbin'
```

Since `path/*` always risks hitting an "argument list too long" error.